modules/siddhi-core/src/main/java/io/siddhi/core/util/collection/operator/IndexOperator.java

```diff
@@ -59,12 +59,14 @@ public boolean contains(StateEvent matchingEvent, Object storeEvents) {
     }
 
     @Override
-    public void delete(ComplexEventChunk<StateEvent> deletingEventChunk, Object storeEvents) {
+    public int delete(ComplexEventChunk<StateEvent> deletingEventChunk, Object storeEvents) {
```

Comment:
Delete should not return the total events but only the deleted number of events

Reply:
it is returning the difference between the number of events before delete and the number after delete